File: backend/app/Models/CreditTransaction.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class CreditTransaction extends Model
{
    protected $fillable = [
        'driver_id',
        'type',
        'amount',
        'balance_before',
        'balance_after',
        'ride_id',
        'description',
    ];

    protected $casts = [
        'amount'         => 'integer',
        'balance_before' => 'integer',
        'balance_after'  => 'integer',
    ];

    public function driver(): BelongsTo
    {
        return $this->belongsTo(User::class, 'driver_id');
    }

    public function ride(): BelongsTo
    {
        return $this->belongsTo(Ride::class);
    }
}

File: mobile/lib/core/models/credit_transaction.dart
class CreditTransaction {
  final int id;
  final String type;
  final int amount;
  final int balanceBefore;
  final int balanceAfter;
  final String? description;
  final int? rideId;
  final DateTime createdAt;

  const CreditTransaction({
    required this.id,
    required this.type,
    required this.amount,
    required this.balanceBefore,
    required this.balanceAfter,
    this.description,
    this.rideId,
    required this.createdAt,
  });

  factory CreditTransaction.fromJson(Map<String, dynamic> json) {
    return CreditTransaction(
      id: json['id'] as int,
      type: json['type'] as String,
      amount: json['amount'] as int,
      balanceBefore: json['balance_before'] as int,
      balanceAfter: json['balance_after'] as int,
      description: json['description'] as String?,
      rideId: json['ride_id'] as int?,
      createdAt: DateTime.parse(json['created_at'] as String),
    );
  }
}
